### DISCUSSION TITLE
Keyboard layer pinning

### DISCUSSION THREAD

**Discussion by argenkiwi** [yqrashawn/GokuRakuJoudo#245] [Q&A]
I would like to be able to implement (sim)[layer pinning](https://argenkiwi.medium.com/keyboard-layer-pinning-20aafede96e5) but I'm struggling to figure out how. In summary, I would like to be able to activate a _simlayer_, hold the spacebar, release the `:key` assigned to the _simlayer_ while keeping it active until the spacebar is released. 

Thanks in advance for any tips you could give me.

**Comment by joshuawagner** [comment]
What a cool idea! I think you'd want to use variables. For example `spaceDown`, `modeF`, `modeJ`, etc. 

---
- description: layer 1 spacebar mode
- `from` spacebar
- `to_if_alone` spacebar
- `to` set variable `spaceDown` to 1
- `to_after_key_up`
  - set variable `spaceDown` to 0
  - set variable `modeF` to 0
  - set variable `modeJ` to 0
---
- description: layer 2 f mode
- `from` F
- `to` set variable `modeF` to 1
- condition: `spaceDown` = 1
---
- description: layer 2 j mode
- `from` J
- `to` set variable `modeJ` to 1
- condition: `spaceDown` = 1
---
- description: f mode, k
- `from` k
- `to` shell script...
- condition: 
  - `modeF` = 1
---
This logic is based on mxstbr's karabiner config that you can see here: https://github.com/mxstbr/karabiner/blob/main/karabiner.json

I don't know if this is possible in Goku.

**Reply by argenkiwi** [reply]
Thanks @joshuawagner! That's very helpful. I would hope Goku allows you to manipulate variables to the same extend, I'll have a play and see if I can make it work on my layout.

**Reply by joshuawagner** [reply]
Please share what you come up with!
